docs/src/intro/overview.md

```diff
@@ -8,7 +8,7 @@ Miden VM consists of three high-level components as illustrated below.
 These components are:
 * **Stack** which is a push-down stack where each item is a field element. Most assembly instructions operate with values located on the stack. The stack can grow up to $2^{32}$ items deep, however, only the top 16 items are directly accessible.
 * **Memory** which is a linear random-access read-write memory. The memory is word-addressable, meaning, four elements are located at each address, and we can read and write elements to/from memory in batches of four. Memory addresses can be in the range $[0, 2^{32})$.
-* **Advice provider** which is a way for the prover to provide nondeterministic inputs to the VM. The advice provider contains a single *advice tape* and unlimited number of *merkle sets*. The latter contain structured data which can be interpreted as a set of Merkle paths.
+* **Advice provider** which is a way for the prover to provide nondeterministic inputs to the VM. The advice provider is composed of a single *advice stack*, an *advice map*, and a *merkle store*. The *advice stack* yields elements to the VM stack; the *advice map* will store temporary, key-mapped stacks that can be written to the VM stack; finally, the merkle store will be the backend for Merkle tree implementations, providing a common interface to persist Merkle data.
```

Comment:
For advice map, I'd probably say it like this:

"the *advice map* stores key-mapped element lists which can be pushed onto the advice stack;"

And for Merkle store, I'd say:

"finally, the Merkle store contains structured Merkle tree data and serves Merkle paths to the VM."

Reply:
Done!